cmd/system-probe/modules/service_discovery_linux.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package modules
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/procfs"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/model"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/portlist"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+const (
+	pathOpenPorts = "/open_ports"
+	pathGetProc   = "/procs/{pid}"
+)
+
+// ServiceDiscoveryModule is the language detection module factory
+var ServiceDiscoveryModule = module.Factory{
+	Name:             config.ServiceDiscoveryModule,
+	ConfigNamespaces: []string{"service_discovery"},
+	Fn: func(_ *sysconfigtypes.Config, _ optional.Option[workloadmeta.Component], _ telemetry.Component) (module.Module, error) {
+		if ddconfig.IsContainerized() {
+			// service_discovery check is not enabled in containerized environments
+			return nil, module.ErrNotEnabled
+		}
+
+		poller, err := portlist.NewPoller()
+		if err != nil {
+			return nil, err
+		}
+		return &serviceDiscovery{
+			portPoller: poller,
+		}, nil
+	},
+	NeedsEBPF: func() bool {
+		return false
+	},
+}
+
+type serviceDiscovery struct {
```

Comment:
I suggest moving this type to a new package `pkg/servicediscovery` (or whatever path you think is appropriate) that has `CODEOWNERS` set to your team, so that code reviews are requested properly. It will also make running tests in Kernel Matrix Testing (KMT) a bit easier.